test-suite/src/test/scala/org/scalajs/testsuite/javalib/NavigableSetTest.scala

```diff
@@ -2,9 +2,124 @@ package org.scalajs.testsuite.javalib
+      expect(nsString.ceiling("z")).toEqual(null)
+    }
+
+    it("should retrieve floor(ordered) elements") {
+    val lInt = asJavaCollection(Set(1, 5, 2, 3, 4))
+      val nsInt = setFactory.empty[Int]
+
+      nsInt.addAll(lInt)
+
+      expect(nsInt.floor(10)).toEqual(5)
+      expect(nsInt.floor(5)).toEqual(5)
+      expect(nsInt.floor(3)).toEqual(3)
+      expect(nsInt.floor(1)).toEqual(1)
+
+      val lString = asJavaCollection(Set("a", "e", "b", "c", "d"))
+      val nsString = setFactory.empty[String]
+
+      nsString.addAll(lString)
+
+      expect(nsString.floor("zzzzz")).toEqual("e")
+      expect(nsString.floor("d")).toEqual("d")
+      expect(nsString.floor("b")).toEqual("b")
+      expect(nsString.floor("a")).toEqual("a")
+      expect(nsString.floor("0")).toEqual(null)
+    }
+
+    it("should retrieve higher(ordered) elements") {
+    val lInt = asJavaCollection(Set(1, 5, 2, 3, 4))
+      val nsInt = setFactory.empty[Int]
+
+      nsInt.addAll(lInt)
+
+      expect(nsInt.higher(4)).toEqual(5)
+      expect(nsInt.higher(3)).toEqual(4)
+      expect(nsInt.higher(1)).toEqual(2)
+      expect(nsInt.higher(-10)).toEqual(1)
+
+      val lString = asJavaCollection(Set("a", "e", "b", "c", "d"))
+      val nsString = setFactory.empty[String]
+
+      nsString.addAll(lString)
+
+      expect(nsString.higher("zzzzz")).toEqual(null)
+      expect(nsString.higher("d")).toEqual("e")
+      expect(nsString.higher("b")).toEqual("c")
+      expect(nsString.higher("a")).toEqual("b")
+      expect(nsString.higher("0")).toEqual("a")
+    }
+
+    it("should retrieve lower(ordered) elements") {
+    val lInt = asJavaCollection(Set(1, 5, 2, 3, 4))
+      val nsInt = setFactory.empty[Int]
+
+      nsInt.addAll(lInt)
+
+      expect(nsInt.lower(5)).toEqual(4)
+      expect(nsInt.lower(4)).toEqual(3)
+      expect(nsInt.lower(3)).toEqual(2)
+      expect(nsInt.lower(10)).toEqual(5)
+
+      val lString = asJavaCollection(Set("a", "e", "b", "c", "d"))
+      val nsString = setFactory.empty[String]
+
+      nsString.addAll(lString)
+
+      expect(nsString.lower("zzzzz")).toEqual("e")
+      expect(nsString.lower("d")).toEqual("c")
+      expect(nsString.lower("b")).toEqual("a")
+      expect(nsString.lower("a")).toEqual(null)
+      expect(nsString.lower("0")).toEqual(null)
+    }
+
+    it("should poll first and last elements") {
+    val lInt = asJavaCollection(Set(1, 5, 2, 3, 4))
+      val ns = setFactory.empty[Int]
+
+      ns.addAll(lInt)
+
+      expect(ns.contains(1)).toBeTruthy
+      expect(ns.pollFirst()).toEqual(1)
+      expect(ns.contains(1)).toBeFalsy
+      expect(ns.pollLast()).toEqual(5)
+      expect(ns.pollFirst()).toEqual(2)
+      expect(ns.pollLast()).toEqual(4)
+      expect(ns.pollFirst()).toEqual(3)
+      expect(ns.isEmpty()).toBeTruthy
+    }
+  }
+}
+
 object NavigableSetFactory {
   def allFactories: Iterator[NavigableSetFactory] =
-    ConcurrentSkipListSetFactory.allFactories
+    Iterator.empty
```

Comment:
You should add it back, this will what happens with the `ConcurrentSkipListSet` when passed to the methods of `java.util.Collections`. This will cover more complex test cases and potentially show bug that happen only with specific combinations of operations.